syntax = "proto3";

package whstorekv;

message HelloRequest {
  string name = 1;
}
message HelloReply {
  string message = 1;
}

message ValueData{
	uint32 ver = 1;
	bytes value = 2;
}

message KVData{
	bytes key = 1;
	ValueData values = 2;
}

message GetReq{
	bytes key  = 1;
}
message GetRsp{
	ValueData data = 1;
}

message SetReq{
	bytes key  = 1;
	ValueData data = 2;
}
message SetRsp{
}

message BatchGetReq{
	repeated bytes keys = 1;
}
message BatchGetRsp{
	repeated KVData datas = 1;
}

message SearchKeysReq{
	bytes key_pre = 1;
	uint32 limit = 2;	
	uint32 order = 3; 
}   

message SearchKeysRsp{
	repeated KVData datas = 1;
}   

service WhStoreKv {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc Get (GetReq) returns (GetRsp) {}
  rpc Set (SetReq) returns (SetRsp) {}
  rpc BatchGet (BatchGetReq) returns (BatchGetRsp) {}
  rpc SearchKeys (SearchKeysReq) returns (SearchKeysRsp) {}
}

